### PR TITLE
Next meetup Section

### PR DESCRIPTION
**Pull request by faqndo97**
CleanShot 2024-04-15 at 02 03 57

# To Do

- [x] Animation
- [x] Final review of styles
- [x] Final responsive review
- [x] Ensure the section is hidden before deploy
- [x] ~ Link to google maps on location

### CODE REVIEW

File: _includes/next_meetup.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{% assign meetup_date = '15 de septiembre' %}
+{% assign meetup_time = '19 hs' %}
+{% assign meetup_place = 'Eagerworks | Juan Paullier 1218' %}
+{% assign meetup_first_speaker_name = 'Marcos Bellucci' %}
+{% assign meetup_first_speaker_talk = 'Ruby on Wasm' %}
+{% assign meetup_second_speaker_name = 'Marcos Bellucci' %}
+{% assign meetup_second_speaker_talk = 'Ruby on Wasm' %}
```
Comment (faqndo97) [inline _includes/next_meetup.html:7]: @santib Do you know if this is the correct way of assigning variables in an "partial"? I think front matter doesn't work here but I'm not super familiar with Jekyll.

File: _includes/next_meetup.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+<section class="next-meetup">
+  <header>
+    <div class="example">
+      <h1>Proxima meet up</h1>
```
Comment (santib) [inline _includes/next_meetup.html:12]:
```suggestion
      <h1>Próxima meetup</h1>
```

File: _includes/next_meetup.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      <svg width="41" height="39" viewBox="0 0 41 39" fill="none" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false">
+        <path d="M20.2242 19.6973L39.8992 23.5392L33.4961 34.9484L20.3406 19.9302L26.6273 38.7902H13.5883L20.1078 19.8138V19.6973H20.2242L20.1078 19.5809L13.5883 0.488037H26.6273L20.4571 19.2317L33.3797 4.3299L40.0156 15.7391L20.3406 19.5809L20.2242 19.6973ZM6.71951 4.3299L19.9914 19.5809L0.316406 15.7391L6.71951 4.3299ZM6.95235 34.9484L0.316406 23.5392L20.1078 19.6973L6.95235 34.9484Z" fill="white"/>
+      </svg>
+    </div>
+  </header>
+  <article>
+    <div class="next-meetup__details">
+      <h2>Podés ser parte de la próxima Meet Up!</h2>
```
Comment (santib) [inline _includes/next_meetup.html:20]:
```suggestion
      <h2>Podés ser parte de la próxima meetup!</h2>
```

File: _includes/next_meetup.html (hunk 1 above)
Comment (santib) [reply _includes/next_meetup.html:7]: I don't have much experience with Jekyll either, but seems that includes files are not intended to define front matter, so I'm not sure if there is a better option than what you did here. @elcuervo might know better

File: _includes/next_meetup.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
+{% assign meetup_date = '15 de septiembre' %}
+{% assign meetup_time = '19 hs' %}
+{% assign meetup_place = 'Eagerworks | Juan Paullier 1218' %}
+{% assign meetup_first_speaker_name = 'Marcos Bellucci' %}
```
Comment (santib) [inline _includes/next_meetup.html:4]: Before merging, maybe we can put "To be confirmed" in each field

Comment (santib) [reply _includes/next_meetup.html:4]: or an if statement to not show the section until we are ready? (or both?) thinking that while we prepare a meetup we might have some things ready before everything is confirmed?

File: assets/js/nav.js [hunk 5]
```diff
@@ -3,3 +3,20 @@ function toggleNavLinks() {
...
+}
+
+document.addEventListener("DOMContentLoaded", () => {
+  const viewportWidth = document.documentElement.clientWidth || window.innerWidth;
+
+  if (viewportWidth > 768) {
+    const element = document.querySelector(".example")
+    duplicate(element, {times: 6})
+  }
+})
```
Comment (santib) [inline assets/js/nav.js:22]: I think we can use CSS for this. Something like

```css
  animation: slide 15s linear infinite;
@keyframes slide {
  0% {
    transform: translateX(-10%);
  }
  100% {
    transform: translateX(-100%);
  }
}
```

then we need to see if repeating the text with CSS or just using a for loop in to replicate the HTML N times.

File: _includes/next_meetup.html (hunk 1 above)
Comment (elcuervo) [reply _includes/next_meetup.html:7]: Another way would be to have this as a JSON data file: https://jekyllrb.com/docs/datafiles/
Same as we are doing with https://github.com/rubyuy/ruby.uy/blob/main/_data/companies.yml.

Comment (faqndo97) [reply _includes/next_meetup.html]: Thanks for the help, I extracted the data into a datafile.

File: _includes/next_meetup.html (hunk 4 above)
Comment (faqndo97) [reply _includes/next_meetup.html]: I implemented both so we can have full control. Thanks for the suggestion.

File: assets/js/nav.js (hunk 5 above)
Comment (faqndo97) [reply assets/js/nav.js:22]: oh the idea is to have an animation? I thought it was just repeated. If the idea is an animation I agree that we can do it with css.

Comment (santib) [reply assets/js/nav.js:22]: Right, I think the idea is that with the animation we catch the attention to the next meetup section 😎

File: _includes/next_meetup.html (hunk 1 above)
Comment (santib) [reply _includes/next_meetup.html]: changes look great 😄

File: _data/next_meetup.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+show: true
+date: 15 de Abril
+time: 19 hrs
+host: rootstrap
+talks:
+  - speaker: guillermo_aguirre
+    name: Ruby on Wasm
+  - speaker: emilio_cristalli
+    name: Ruby on Wasm
```
Comment (santib) [inline _data/next_meetup.yml:9]: great to have this here so we can easily check in the review app but leaving a reminder here to update it before merge.

Comment (faqndo97) [reply _data/next_meetup.yml:9]: Added a todo for this, thanks!

File: _includes/next_meetup.html [hunk 7]
```diff
@@ -0,0 +1,58 @@
+{% assign next_meetup = site.data.next_meetup %}
+{% assign host = site.data.companies[next_meetup.host] %}
+
+{% if next_meetup.show %}
+  <section class="next-meetup">
+    <header>
+      <div class="next-meetup-callout" aria-hidden=true>
+        <h1 class="next-meetup-callout__title">Proxima meetup</h1>
```
Comment (santib) [inline _includes/next_meetup.html:8]:
```suggestion
        <h1 class="next-meetup-callout__title">Próxima meetup</h1>
```

**Pull request merged**